File: src/adapter/MallardDuck.java
package adapter;

/**
 * Created by bvvy on 2017/7/20.
 */
public class MallardDuck implements Duck {
    @Override
    public void quack() {
        System.out.println("嘎嘎嘎");
    }

    @Override
    public void fly() {
        System.out.println("i m flying");
    }
}

File: src/strategy/fail_design/MallardDuck.kt
package strategy.fail_design

/**
 * Created by bvvy on 2017/7/13.
 */
open class MallardDuck : Duck() {

    override fun quack() {
        super.quack()
    }

    override fun swim() {
        super.swim()
    }

    override fun display() {
        print("I have a green head")
    }

    override fun fly() {
        super.fly()
    }
}